bitwarden_access_manager.py

```diff
@@ -0,0 +1,672 @@
+            case 2:
+                type="user"
+            case 3:
+                type="manager"
+            case 4:
+                type="custom"
+        return type
+
+    def get_members(self) -> Iterable[Member]:
+        data = self._http_get(f"/public/members")
+        members= json.load(data)
+
+        for member in members["data"]:
+            type=self.set_member_type(member["type"])
+
+            yield Member(
+                id=member["id"],
+                name=member["name"],
+                email=member["email"],
+                type=type,
+                access_all=member["accessAll"],
+            )
+
+class Configuration(NamedTuple):
+    collection: Set[Collection]
+    member: Set[Member]
+    group: Set[Group]
+    group_memberships: Set[GroupMember]
+
+    @staticmethod
+    def from_toml_dict(data: Dict[str, Any]) -> Configuration:
+        collection = {Collection.from_toml_dict(c) for c in data["collection"]}
+        member = {Member.from_toml_dict(m) for m in data["member"]}
+        group = {Group.from_toml_dict(m) for m in data["group"]}
+        group_memberships = {
+            GroupMember(
+                member_id=member["member_id"],
+                member_name=member["member_name"],
+                group_name=group,
+            )
+            for member in data["member"]
+            for group in member.get("groups", [])
+        }
+        return Configuration(
+            collection=collection,
+            member=member,
+            group=group,
+            group_memberships=group_memberships,
+        )
+
+    @staticmethod
+    def from_toml_file(fname: str) -> Configuration:
+        with open(fname, "rb") as f:
+            data = tomllib.load(f)
+            return Configuration.from_toml_dict(data)
+
+def print_indented(lines: str) -> None:
+    """Print the input indented by two spaces."""
+    for line in lines.splitlines():
+        print(f"  {line}")
+
+
+def print_simple_diff(actual: str, target: str) -> None:
+    """
+    Print a line-based diff of the two strings, without abbreviating large
+    chunks of identical lines like a standard unified diff would do.
+    """
+    lines_actual = actual.splitlines()
+    lines_target = target.splitlines()
+    line_diff = SequenceMatcher(None, lines_actual, lines_target)
+    for tag, i1, i2, j1, j2 in line_diff.get_opcodes():
+        if tag == "equal":
+            for line in lines_actual[i1:i2]:
+                print("  " + line)
+        elif tag == "replace":
+            for line in lines_actual[i1:i2]:
+                print("- " + line)
+            for line in lines_target[j1:j2]:
+                print("+ " + line)
+        elif tag == "delete":
+            for line in lines_actual[i1:i2]:
+                print("- " + line)
+        elif tag == "insert":
+            for line in lines_target[j1:j2]:
+                print("+ " + line)
+        else:
+            raise Exception("Invalid diff operation.")
```

Comment:
I suppose this is just copied verbatim from the GitHub script? For now that’s okay, we can unify them later, it’s just that I did not review this function in detail now.

Reply:
Yes this is a copy from the Github script.